### DISCUSSION TITLE
Automating DECK.GL with Selenium

### DISCUSSION THREAD

**Discussion by pranav165** [visgl/deck.gl#6016] [Q&A]
Hi 
Recently our development team has moved to deck.gl . Since it does not expose any html elements that we can locate using Selenium , could someone help
how can we perform click etc operations on the elements with deck.gl .

**Comment by Pessimistress** [comment]
You can select the canvas using [id](https://deck.gl/docs/api-reference/core/deck#id).

**Reply by pranav165** [reply]
But how to work with elements within canvas. like click on something inside canvas

**Reply by Pessimistress** [reply]
You cannot select objects in a WebGL context like how you select DOM objects. You need to figure out where things are rendered and click on a specific pixel.
